src/structures/interfaces/InteractionResponses.js

```diff
@@ -0,0 +1,229 @@
+'use strict';
+
+const WebhookClient = require('../../client/WebhookClient');
+const { InteractionResponseTypes } = require('../../util/Constants');
+const MessageFlags = require('../../util/MessageFlags');
+const APIMessage = require('../APIMessage');
+
+/**
+ * Interface for classes that support shared interaction response types.
+ * @interface
+ */
+class InteractionResponses {
+  constructor() {
+    /**
+     * Whether the reply to this interaction has been deferred
+     * @type {boolean}
+     */
+    this.deferred = false;
+
+    /**
+     * Whether this interaction has already been replied to
+     * @type {boolean}
+     */
+    this.replied = false;
+
+    /**
+     * An associated webhook client, can be used to create deferred replies
+     * @type {WebhookClient}
+     */
+    this.webhook = new WebhookClient(this.applicationID, this.token, this.client.options);
```

Comment:
Are the properties, specially this one, required to exist? The entire class is a mixin and is not subclassed, and besides, it reads properties that do not exist.

Reply:
When the interface is applied, the `this` not longer refers to the interface's, and the constructor is never copied for obvious reasons, so those lines are never called.

Reply:
This structuring does exist on other interfaces in the library, not exactly sure why. I always thought of it as a reminder for lib devs kinda like "hey you need this prop for this interface to work"